scene/main/viewport.cpp

```diff
@@ -3467,17 +3468,62 @@ bool Viewport::is_using_xr() {
 	return use_xr;
 }
 
-void Viewport::set_scale_3d(float p_scale_3d) {
+void Viewport::set_scaling_3d_mode(Scaling3DMode p_scaling_3d_mode) {
+	if (scaling_3d_mode == p_scaling_3d_mode) {
+		return;
+	}
+
+	scaling_3d_mode = p_scaling_3d_mode;
+	RS::get_singleton()->viewport_set_scaling_3d_mode(viewport, (RS::ViewportScaling3DMode)(int)p_scaling_3d_mode);
+}
+
+Viewport::Scaling3DMode Viewport::get_scaling_3d_mode() const {
+	return scaling_3d_mode;
+}
+
+void Viewport::set_scaling_3d_quality(float p_scaling_3d_quality) {
```

Comment:
Should be renamed to ``scaling_3d_scale`` as it is in ProjectSettings and as it is exposed to users.